bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionScheduler.java

```diff
@@ -0,0 +1,85 @@
+/*-
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2013 - 2022 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.packagegarbagecollector;
+
+import com.adobe.acs.commons.util.RequireAem;
+import org.apache.sling.event.jobs.JobManager;
+import org.apache.sling.event.jobs.ScheduledJobInfo;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component(
+        immediate = true,
+        configurationPolicy = ConfigurationPolicy.REQUIRE
+)
+@Designate(ocd = PackageGarbageCollectionConfig.class, factory = true)
+public class PackageGarbageCollectionScheduler {
+    private static final Logger LOG = LoggerFactory.getLogger(PackageGarbageCollectionScheduler.class);
+    public static final String JOB_TOPIC = "com/adobe/acs/commons/PackageGarbageCollectionJob";
+    public static final String GROUP_NAME = "groupName";
+    public static final String MAX_AGE_IN_DAYS = "maxAgeInDays";
+
+    @Reference
+    JobManager jobManager;
+
+    @Reference(target="(distribution=classic)")
+    RequireAem requireAem;
+
+    ScheduledJobInfo job;
+
+    @Activate
+    protected void activate(PackageGarbageCollectionConfig config) {
+        job = scheduleJob(config);
```

Comment:
What happens if you have the system running and it does terminate without proper shutdown? I think you will end up with the same job running twice after the next restart.
So I would check on startup if such a job is already present and unregister all of them. And then register a new one. (That's probably easier to implement than checking existing jobs if they have the correct schedule).
This also improves the handling in case you are running on a cluster.

I recently updated the Sling documentation covering this aspect: https://sling.apache.org/documentation/bundles/apache-sling-eventing-and-job-handling.html#scheduled-jobs-1